projects/noloan/app/src/main/java/hasadna/noloan/DbMessages.java

```diff
@@ -40,45 +64,74 @@ public void init(List<SmsMessage> messages) {
 
   // Checks if message had already been suggested. Updates counter / Adds a new suggestion
   public void suggestMessage(SmsMessage smsMessage) {
-
-    // Search if message was already suggested -> Modify counter (+1)
+    // Case: Message was already suggested by others but not by this user
+    // 1. Add user as a "suggester"
     int index = searchMessage(smsMessage);
-    if (index != -1) {
+    if ((index != -1) && !messages.get(index).getSuggestersList().contains(firebaseUser.getUid())) {
       SmsMessage newMessage =
-          messages.get(index).toBuilder().setCounter(messages.get(index).getCounter() + 1).build();
+          messages
+              .get(index)
+              .toBuilder()
+              .addSuggesters(firebaseUser.getUid())
+              // 2. Update counter
+              .setCounter(messages.get(index).getCounter() + 1)
+              .setId(messages.get(index).getId())
+              .build();
       firestoreClient.modifyMessage(messages.get(index), newMessage);
       modifyMessage(newMessage);
-      if (messagesListener != null) {
-        messagesListener.messageModified(index);
-      }
+
+      // Notify
+      notifyListeners(messages.indexOf(newMessage), newMessage, Type.MODIFIED);
+
     }
-    // Add new suggestion
-    else {
+    // Case New suggestion
+    else if (index == -1) {
       firestoreClient.writeMessage(
-          smsMessage.toBuilder().setCounter(smsMessage.getCounter() + 1).build());
+          smsMessage
+              .toBuilder()
+              .setCounter(smsMessage.getCounter() + 1)
+              .addSuggesters(firebaseUser.getUid())
+              .build());
     }
   }
 
   public void undoSuggestion(SmsMessage smsMessage) {
 
-    // If other people suggested this message as well - Modify message with a new counter (-1)
-    if (smsMessage.getCounter() > 1) {
-      SmsMessage newMessage =
-          smsMessage.toBuilder().setCounter(smsMessage.getCounter() - 1).build();
-      firestoreClient.modifyMessage(smsMessage, newMessage);
-      modifyMessage(newMessage);
-      if (messagesListener != null) {
-        messagesListener.messageModified(messages.indexOf(smsMessage));
+    // Check if user is part of the "suggesters" of this spam message
+    if (smsMessage.getSuggestersList().contains(firebaseUser.getUid())) {
+
+      // Case: Other people had suggested this spam as well, update just the counter (-1)
+      if (smsMessage.getCounter() > 1) {
```

Comment:
Why do you need a counter, now that you can use `smsMessage.getSuggestersList().getSize()`?

Reply:
Thanks, I was wondering about this one.
So perhaps preferably to remove the Proto field `counter`, and just use `SmsMessage.getSuggestersCount()` for this purpose yes?

Reply:
Yup.